<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/urdf/schunk_gripper_jpg_p_160_pybullet.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- It is required to pass the args: robot_name-->
<robot name="schunk_gripper_jpg_p_160">
  <link concave="yes" name="gripper_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0.00002849 -0.00022471 0.03657638"/>
      <mass value="0.6818"/>
      <inertia ixx="0.00154456" ixy="0.00000154" ixz="0.00000118" iyy="0.00283260" iyz="-0.00000405" izz="0.00195998"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_base.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision concave="yes">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_base.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="left_joint" type="prismatic">
    <origin rpy="0 0 0" xyz="0.0006 0 0.0481"/>
    <parent link="gripper_base"/>
    <child link="left_slider"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0.000" upper="0.017" velocity="1000000"/>
  </joint>
  <link concave="yes" name="left_slider">
    <inertial>
      <origin rpy="0 0 0" xyz="0.03936113 -0.00000000 0.01237203"/>
      <mass value="0.070"/>
      <inertia ixx="0.00002269" ixy="0.00000000" ixz="0.00003426" iyy="0.00015861" iyz="-0.00000000" izz="0.00015142"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_finger_slider.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision concave="yes">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_finger_slider.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="shadow_left_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="left_slider"/>
    <child link="shadow_left_slider"/>
  </joint>
  <link name="shadow_left_slider">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0" ixz="0.0" iyy="0.00001" iyz="0.0" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="right_joint" type="prismatic">
    <origin rpy="0 0 3.14159" xyz="-0.0006 0 0.0481"/>
    <parent link="gripper_base"/>
    <child link="right_slider"/>
    <axis xyz="1 0 0"/>
    <limit effort="100000" lower="0.000" upper="0.017" velocity="1000000"/>
    <mimic joint="left_joint" multiplier="1" offset="0"/>
  </joint>
  <link concave="yes" name="right_slider">
    <inertial>
      <origin rpy="0 0 0" xyz="0.03936113 -0.00000000 0.01237203"/>
      <mass value="0.070"/>
      <inertia ixx="0.00002269" ixy="0.00000000" ixz="0.00003426" iyy="0.00015861" iyz="-0.00000000" izz="0.00015142"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_finger_slider.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision concave="yes">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/SCHUNK_parallel_gripper_finger_slider.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="shadow_right_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="right_slider"/>
    <child link="shadow_right_slider"/>
  </joint>
  <link name="shadow_right_slider">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0" ixz="0.0" iyy="0.00001" iyz="0.0" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="left_finger_joint" type="fixed">
    <origin rpy="0 0 3.14159" xyz="0.0285 0 0.0289"/>
    <parent link="left_slider"/>
    <child link="left_finger"/>
    <axis xyz="0 0 0"/>
  </joint>
  <link concave="yes" name="left_finger">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01665438 -0.00000438 0.05348249"/>
      <mass value="0.10"/>
      <inertia ixx="0.00041666" ixy="-0.00000001" ixz="-0.00005970" iyy="0.00047996" iyz="-0.00000004" izz="0.00007666"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/Finger.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision concave="yes">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/Finger.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="right_finger_joint" type="fixed">
    <origin rpy="0 0 3.14159" xyz="0.0285 0 0.0289"/>
    <parent link="right_slider"/>
    <child link="right_finger"/>
    <axis xyz="0 0 0"/>
  </joint>
  <link concave="yes" name="right_finger">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01665438 -0.00000438 0.05348249"/>
      <mass value="0.10"/>
      <inertia ixx="0.00041666" ixy="-0.00000001" ixz="-0.00005970" iyy="0.00047996" iyz="-0.00000004" izz="0.00007666"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/Finger.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision concave="yes">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/gauss/projects/personal_ws/src/schunk_gripper_jpg_p_160/schunk_gripper_jpg_p_160_description/meshes/Finger.STL"/>
      </geometry>
    </collision>
  </link>
  <transmission name="left_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_joint_circle">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="right_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_joint_circle">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <joint name="left_joint_circle" type="revolute">
    <origin rpy="0 0.0 0" xyz="0.0 0.0 0.0"/>
    <parent link="gripper_base"/>
    <child link="left_circle"/>
    <axis xyz="0 1 0"/>
    <limit effort="10000" lower="-100" upper="100" velocity="1000"/>
  </joint>
  <link name="left_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
  </link>
  <joint name="middle_left_joint_circle" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="left_circle"/>
    <child link="middle_left_circle"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-1" upper="1" velocity="1000"/>
  </joint>
  <link name="middle_left_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
  </link>
  <joint name="end_left_joint_circle" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <parent link="middle_left_circle"/>
    <child link="end_left_circle"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-200" upper="200" velocity="1000"/>
  </joint>
  <link name="end_left_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.1 0.01"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
  </link>
  <joint name="right_joint_circle" type="revolute">
    <origin rpy="0 0.0 0" xyz="0.0 0.0 0.0"/>
    <parent link="gripper_base"/>
    <child link="right_circle"/>
    <axis xyz="0 1 0"/>
    <limit effort="10000" lower="-100" upper="100" velocity="1000"/>
  </joint>
  <link name="right_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
  </link>
  <joint name="middle_right_joint_circle" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="right_circle"/>
    <child link="middle_right_circle"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-1" upper="1" velocity="1000"/>
  </joint>
  <link name="middle_right_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
  </link>
  <joint name="end_right_joint_circle" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <parent link="middle_right_circle"/>
    <child link="end_right_circle"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-200" upper="200" velocity="1000"/>
  </joint>
  <link name="end_right_circle">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <mass value="0.0001"/>
      <inertia ixx="0.000001" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="-0.0" izz="0.000001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.1 0.01"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
  </link>
</robot>

